src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1956,7 +1955,22 @@ void Session::send_bind_message()
     OutputBuffer& out = m_conn.get_output_buffer();
     // Discard the token since it's ignored by the server.
     std::string empty_access_token{};
-    protocol.make_bind_message(protocol_version, out, session_ident, path, empty_access_token, need_client_file_ident,
+    std::string path_data;
+    if (m_is_flx_sync_session) {
+        nlohmann::json bind_json_data;
+        if (protocol_version >= 8) {
```

Comment:
I think this kind of protocol version specific logic should really go in the protocol codec. It's the thing that decides what version/mode of the wire protocol results in which bytes get put on the wire.

Reply:
I actually had it as two different functions with the FLX one taking a json object, but changed it to this since all the other functions just took strings. I'll change it back.

Reply:
Updated